.travis.yml

```diff
@@ -50,13 +57,29 @@ script:
         mkdir -p "${HOME}/.foldercache/${exercise}/.stack-work"
         ln -f -s "${HOME}/.foldercache/${exercise}/.stack-work"
 
-        # Here we prepare the exercise to be tested by Stack.
-        MODULE=`sed -n 's/ *module \+\([a-zA-Z0-9]\+\).*/\1/p' src/Example.hs`
-        mv src/Example.hs "src/${MODULE}.hs"
+        if [ -f src/Example.hs ]; then
+            # Here we prepare the exercise to be tested by Stack.
+            MODULE=`sed -n 's/ *module \+\([a-zA-Z0-9]\+\).*/\1/p' src/Example.hs`
+            mv src/Example.hs "src/${MODULE}.hs"
+
+            test_exercise
+        else
+            found=
+            for example in examples/*; do
+                echo "testing ${example}"
+                rm -f src/*.hs
+                mv ${example}/*.hs src
+                mv ${example}/package.yaml .
+                found=1
+
+                test_exercise
+            done
+
+            if [ -z "$found" ]; then
```

Comment:
Found one link about it, but I couldn't test test it yet.
http://stackoverflow.com/questions/2937407/test-whether-a-glob-has-any-matches-in-bash